### PR TITLE
Add click and collect and preorder order filters

### PR DESCRIPTION
**Pull request by IKarbowiak**
Add order filters:
- `isClickAndCollect`
- `isPreorder`

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-4851 Add missing order filters in API](https://app.clickup.com/t/2549495/SALEOR-4851)

File: saleor/graphql/order/filters.py [hunk 1]
```diff
@@ -109,6 +111,32 @@ def filter_channels(qs, _, values):
...
+        variants = ProductVariant.objects.filter(
+            Q(is_preorder=True)
+            & (
+                Q(preorder_end_date__isnull=True)
+                | Q(preorder_end_date__gte=timezone.now())
+            )
+        )
+        lines = OrderLine.objects.filter(
+            Exists(variants.filter(id=OuterRef("variant_id")))
+        )
```
Comment (tomaszszymanski129) [inline saleor/graphql/order/filters.py:134]: `.values("id")`

File: saleor/graphql/order/tests/test_order.py [hunk 2]
```diff
@@ -6784,6 +6785,133 @@ def test_draft_order_query_with_filter_created_(
...
+
+    # when
+    response = staff_api_client.post_graphql(
+        orders_query_with_filter, variables, permissions=(permission_manage_orders,)
+    )
+
+    # then
+    content = get_graphql_content(response)
+    returned_orders = content["data"]["orders"]["edges"]
+    if is_click_and_collect:
```
Comment (L3str4nge) [inline saleor/graphql/order/tests/test_order.py:6823]: Don't we want to create two separate tests?

File: saleor/graphql/order/tests/test_order.py [hunk 3]
```diff
@@ -6784,6 +6785,133 @@ def test_draft_order_query_with_filter_created_(
...
+
+    # when
+    response = staff_api_client.post_graphql(
+        orders_query_with_filter, variables, permissions=(permission_manage_orders,)
+    )
+
+    # then
+    content = get_graphql_content(response)
+    returned_orders = content["data"]["orders"]["edges"]
+    if is_preorder:
```
Comment (L3str4nge) [inline saleor/graphql/order/tests/test_order.py:6900]: Don't we want to create two separate tests?

Comment (IKarbowiak) [reply saleor/graphql/order/tests/test_order.py:6900]: Same here.

File: saleor/graphql/order/tests/test_order.py (hunk 2 above)
Comment (patrys) [reply saleor/graphql/order/tests/test_order.py:6823]: I'm with @L3str4nge here. Conditional logic inside the test body is the best way to tell multiple test cases are pretending to be one.

Comment (L3str4nge) [reply saleor/graphql/order/tests/test_order.py:6823]: Yeah but `given` and `when` could be easily wrapped as `fixture` and reuse. Actually this test does not test one thing but two separate scenarios.

**Review by tomaszszymanski129 — APPROVED** [review]

**Review by L3str4nge — APPROVED** [review]

**Review by korycins — APPROVED** [review]

**Pull request merged**